x/merkledb/db.go

```diff
@@ -1023,17 +1023,21 @@ func (db *merkleDB) applyChanges(ctx context.Context, valueNodeBatch *valueNodeB
 		}
 
 		if shouldAddValue {
-			valueNodeBatch.Put(key, nodeChange.after)
+			if err := db.valueNodeDB.Write(valueNodeBatch, key, nodeChange.after); err != nil {
+				return err
+			}
 		} else if shouldDeleteValue {
-			valueNodeBatch.Delete(key)
+			if err := db.valueNodeDB.Write(valueNodeBatch, key, nil); err != nil {
```

Comment:
i believe nodechange.after should be nil here so i'm not sure this needs to be split anymore.

Reply:
Nvm, this is deletion from the value db so a node could still exist, but with no value.

Reply:
`nodeChange.before` may have a value but `nodeChange.after` might not have a value